### PR TITLE
Fix (nearly) all stage2 crashes when testing stdlib

### PR DESCRIPTION
**Pull request by Vexu**
After this passing all std tests will "only" require a bunch of miscompilation fixes, implementing async, and packed struct related adjustments.
Probably worth noting that removing stage2 function pointer handling from the compress related structs seems to cause #10886.

Should I disable the 40 failing tests and add `test-std` to the CI?

### CODE REVIEW

File: lib/std/compress/deflate/compressor_test.zig [hunk 1]
```diff
@@ -397,8 +396,7 @@ test "deflate/inflate string" {
     // Skip wasi because it does not support std.fs.openDirAbsolute()
     if (builtin.os.tag == .wasi) return error.SkipZigTest;
```
Comment (zeroZshadow) [inline lib/std/compress/deflate/compressor_test.zig:397]: This this skip still stay now that `std.fs.openDirAbsolute()` is no longer used here?

**Comment by andrewrk** [comment]
> Should I disable the 40 failing tests and add test-std to the CI?

Yes. It's time.

**Comment by andrewrk** [comment]
> CI sourcehut: cd out of build before running tests

What's the motivation here? All `zig build` commands are supposed to work from any sub-directory.

**Comment by Vexu** [comment]
> What's the motivation here? All `zig build` commands are supposed to work from any sub-directory.

I thought `@src` in stage2 returns `--main-pkg-path` relative paths but with further testing it seems relative to the Zig file given as input and the correct fix is to make it always return absolute paths.

**Pull request merged**